Sources/ResourceBuilder.swift

```diff
@@ -0,0 +1,233 @@
+// ResourceBuilder.swift
+//
+// The MIT License (MIT)
+//
+// Copyright (c) 2015 Zewo
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+public final class ResourceBuilder: RouterBuilder {}
+
+extension ResourceBuilder {
+    public func get(middleware: Middleware..., respond: Respond) {
+        addRoute(method: .get, middleware: middleware, respond: respond)
+    }
+
+    public func get<I: ResourceIdentifier>(middleware: Middleware..., id: I.Type = I.self, respond: (request: Request, id: I) throws -> Response) {
+        addRoute(method: .get, middleware: middleware, id: id, respond: respond)
+    }
+}
+
+extension ResourceBuilder {
+    public func head(middleware: Middleware..., respond: Respond) {
+        addRoute(method: .head, middleware: middleware, respond: respond)
+    }
+
+    public func head<I: ResourceIdentifier>(middleware: Middleware..., id: I.Type = I.self, respond: (request: Request, id: I) throws -> Response) {
+        addRoute(method: .head, middleware: middleware, id: id, respond: respond)
+    }
+}
+
+extension ResourceBuilder {
+    public func post(middleware: Middleware..., respond: Respond) {
+        addRoute(method: .post, middleware: middleware, respond: respond)
+    }
+
+    public func post(middleware: Middleware..., content: StructuredData.Type, respond: (request: Request, content: StructuredData) throws -> Response) {
+        addRoute(method: .post, middleware: middleware, respond: respond)
```

Comment:
Why do you have to pass in `StructuredData.Type`? Seems like boilerplate to me. Isn't just stating it in the closure enough?
```swift
resource.post { (request: Request, content: StructuredData) in 
    ... 
}
```